public/app/features/plugins/admin/helpers.ts

```diff
@@ -343,3 +349,31 @@ function isDisabledSecretsPlugin(type?: PluginType): boolean {
 export function isLocalCorePlugin(local?: LocalPlugin): boolean {
   return Boolean(local?.signature === 'internal');
 }
+
+function getId(inputString: string): string {
+  const parts = inputString.split(' - ');
+  return parts[0];
+}
+
+export function getPluginDetailsForFuzzySearch(plugins: CatalogPlugin[]): { [key: string]: string } {
+  return plugins.reduce((result: { [key: string]: string }, { id, name, type, orgName, info }: CatalogPlugin) => {
+    const keywordsForSearch = info.keywords?.join(' ').toLowerCase();
+    result[id] = `${id} - ${name} - ${type} - ${orgName} - ${keywordsForSearch}`;
+    return result;
+  }, {});
+}
+export function fuzzySearch(query: string, dataArray: string[]) {
+  let opts = {};
+  let uf = new uFuzzy(opts);
+  let idxs = uf.filter(dataArray, query);
+
+  if (idxs != null && idxs.length > 0) {
```

Comment:
Nit: Just for the fun of it. there is a simpler way to write all this code:

```typescript
if (ids === null) {
    return null
}
return idsx.map( id => getId(dataArray[id]));
```

This will not generate the exact array you were generating which was an object with the key being the whole search string and the value the plugin id, but instead will return an array of only plugin-ids which is easier to use for your case.